### PR TITLE
Update e2e_test go and ginkgo

### PR DESCRIPTION
**Pull request by mrogers950**
Update the e2e_test module go to 1.17 and bump ginkgo/gomega to align with https://github.com/linode/linode-cloud-controller-manager/pull/110.

### General:

* [ ] Have you removed all sensitive information, including but not limited to access keys and passwords?
* [ ] Have you checked to ensure there aren't other open or closed [Pull Requests](../../pulls) for the same bug/feature/question?

### Pull Request Guidelines:

1. [ ] Does your submission pass tests?
1. [ ] Have you added tests? 
1. [ ] Are you addressing a single feature in this PR? 
1. [ ] Are your commits atomic, addressing one change per commit?
1. [ ] Are you following the conventions of the language? 
1. [ ] Have you saved your large formatting changes for a different PR, so we can focus on your work?
1. [ ] Have you explained your rationale for why this feature is needed? 
1. [ ] Have you linked your PR to an [open issue](https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/)

### CODE REVIEW

**Comment by mrogers950** [comment]
I've confirmed that the tests pass with this change.

File: e2e/test/scripts/create_cluster.sh [hunk 1]
```diff
@@ -7,7 +7,8 @@ set -x
 
 export LINODE_API_TOKEN="$1"
 export CLUSTER_NAME="$2"
-export K8S_VERSION="$3"
+export REGION="$3"
+export K8S_VERSION="$4"
```
Comment (git-it127) [inline e2e/test/scripts/create_cluster.sh:11]: In order to not break any possible existing automation I think we should try to keep the params in the same order. So we can try this instead.

```suggestion
export K8S_VERSION="$3"
export REGION="$4"
```

File: e2e/test/scripts/create_cluster.sh [hunk 2]
```diff
@@ -22,7 +23,7 @@ variable "server_type_master" {
   default = "g6-standard-2"
 }
 variable "region" {
-  default = "eu-west"
+  default = "$REGION"
```
Comment (git-it127) [inline e2e/test/scripts/create_cluster.sh:26]: Again to not break any existing automation maybe we can look to check if `REGION` is not set. If not set then keep this default. If it is set then override it with what was passed in.

**Review by git-it127 — CHANGES_REQUESTED** [review]
Overall looks like some good updates. Have a few small suggestions here.

File: e2e/test/scripts/create_cluster.sh (hunk 2 above)
Comment (mrogers950) [reply e2e/test/scripts/create_cluster.sh:32]: Done, thanks for the suggestion.

**Comment by mrogers950** [comment]
@git-it127 I've updated with your suggestions.

**Review by git-it127 — APPROVED** [review]

**Pull request merged**